main.go

```diff
@@ -23,6 +26,21 @@ var (
 )
 
 func main() {
+
+	signalChannel := make(chan os.Signal, 2)
```

Comment:
What's the reason behind buffered channel?

Reply:
Sorry for me being stupid ( I did copy paste from internet) How should I init the buffer instead?

Reply:
Leave the `, 2` part out when buffering is not needed. I'm just wondering if there's something in the implementation details why it needs this. It can be crucial, but I don't know, that's why I asked. :slightly_smiling_face: 

This is the difference between buffered and non-buffered channel: https://play.golang.org/p/HuEYmyf3bbW